src/botPage/view/blockly/blocks/trade/components.js

```diff
@@ -61,11 +61,17 @@ export const duration = block => {
 
 export const payout = block => {
     if (!block.getInput('AMOUNT')) {
-        block
-            .appendValueInput('AMOUNT')
-            .setCheck('Number')
-            .appendField(`${translate('Stake')}:`)
-            .appendField(new Blockly.FieldDropdown(config.lists.CURRENCY), 'CURRENCY_LIST');
+        const amountInput = block.appendValueInput('AMOUNT');
+
+        amountInput.setCheck('Number');
+
+        if (block.type === 'tradeOptions_payout') {
```

Comment:
@aaron-binary should we hardcode types?

Reply:
Using types we can uniquely identify specific blocks, this function will only be called by `tradeOptions-*`-type blocks.